### PR TITLE
Update code references to pegasus teacher-dashboard

### PR DESCRIPTION
**Pull request by Erin007**
In 2019 we transitioned the teacher dashboard off of code.org (Pegasus) and onto studio.code.org (Code Studio).  We left in a redirect from code.org/teacher-dashboard to studio.code.org/teacher_dashboard, given that it's been years since the teacher dashboard content lived on Pegasus I'd like to work toward fully deprecating the Pegasus teacher dashboard. 

I've outlined the next steps in [LP-2153](https://codedotorg.atlassian.net/browse/LP-2153)

This initial PR (sub-task [LP-2154](https://codedotorg.atlassian.net/browse/LP-2154)) begins removing old references in the code to the code.org teacher dashboard and related clean-up. 

- removes code.org/teacher-dashboard/section_progress haml file from the exceptions to linting because that file no longer exists 
- updates old test data urls to the current teacher dashboard url and shares them between test files 
- removes teacher-dashboard route from Pegasus cache special rules 
- updates the `teacher_dashboard_url` helper method 
- updates the base url for section data to point to Studio rather than code.org 
- removes a test checking that students don't see the pegasus link in their user menu, because no one sees that link in their user menu 🙃
- updates a test that checks the route helper to a current pegasus route

### CODE REVIEW

File: cookbooks/cdo-varnish/libraries/http_cache.rb [hunk 1]
```diff
@@ -139,7 +139,6 @@ def self.config(env)
               /amazon-future-engineer*
               /create-company-profile*
               /edit-company-profile*
-              /teacher-dashboard*
```
Comment (Erin007) [inline cookbooks/cdo-varnish/libraries/http_cache.rb:142]: @Hamms this is delayed and partial follow up to the conversation we had about potential ways to clean up the special casing in `http_cache`. What do you think?

File: dashboard/app/helpers/application_helper.rb [hunk 2]
```diff
@@ -112,7 +112,7 @@ def home_url
   end
 
   def teacher_dashboard_url
```
Comment (jamescodeorg) [inline dashboard/app/helpers/application_helper.rb]: My quick search didn't turn up any references to this method, maybe we can just remove it?

File: dashboard/app/models/sections/section.rb [hunk 3]
```diff
@@ -283,7 +283,7 @@ def summarize(include_students: true)
       currentUnitTitle: title_of_current_unit,
       linkToCurrentUnit: link_to_current_unit,
       numberOfStudents: num_students,
-      linkToStudents: "#{base_url}#{id}/manage",
+      linkToStudents: "#{base_url}#{id}/manage_students",
```
Comment (jamescodeorg) [inline dashboard/app/models/sections/section.rb:286]: Do you know if this link is used / visible anywhere?  I'm wondering if a teacher might have recently saved the old link, in which case we might want to ask if it's ok for it to "suddenly" stop working.

**Review by jamescodeorg — APPROVED** [review]
Love that we're doing this cleanup!

File: dashboard/app/models/sections/section.rb (hunk 3 above)
Comment (Erin007) [reply dashboard/app/models/sections/section.rb:286]: We're not linking directly here from anywhere I can find on the site. It's possible that someone who's been teaching with Code.org for a long time (2+ years) has this link bookmarked and always goes through the redirect to get to the current teacher dashboard.

File: dashboard/app/helpers/application_helper.rb (hunk 2 above)
Comment (Erin007) [reply dashboard/app/helpers/application_helper.rb]: yay! I love deleting unused things!

**Pull request merged**